sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3598,15 +3598,47 @@ object GpuOverrides extends Logging {
         TypeSig.STRING,
         Seq(ParamCheck("struct",
           (TypeSig.BOOLEAN + TypeSig.STRING + TypeSig.integral + TypeSig.FLOAT +
-            TypeSig.DOUBLE + TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP).nested(),
+            TypeSig.DOUBLE + TypeSig.DATE + TypeSig.TIMESTAMP +
+            TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP).nested(),
           (TypeSig.BOOLEAN + TypeSig.STRING + TypeSig.integral + TypeSig.FLOAT +
-            TypeSig.DOUBLE + TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP).nested()
+            TypeSig.DOUBLE + TypeSig.DATE + TypeSig.TIMESTAMP +
+            TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP).nested()
         ))),
       (a, conf, p, r) => new UnaryExprMeta[StructsToJson](a, conf, p, r) {
         override def tagExprForGpu(): Unit = {
           if (a.options.get("pretty").exists(_.equalsIgnoreCase("true"))) {
             willNotWorkOnGpu("to_json option pretty=true is not supported")
           }
+          val hasDates = TrampolineUtil.dataTypeExistsRecursively(a.child.dataType,
+            _.isInstanceOf[DateType])
+          if (hasDates) {
+            // check if the default format is being used
+            val defaultFormat = "yyyy-MM-dd"
+            val dateFormat = a.options.getOrElse("dateFormat", defaultFormat)
+            if (dateFormat != defaultFormat) {
+              // we can likely support other formats but we would need to add tests
+              // tracking issue is https://github.com/NVIDIA/spark-rapids/issues/9602
+              willNotWorkOnGpu(s"Unsupported dateFormat '$dateFormat' in to_json")
+            }
+          }
+          val hasTimestamps = TrampolineUtil.dataTypeExistsRecursively(a.child.dataType,
+            _.isInstanceOf[TimestampType])
+          if (hasTimestamps) {
+            // check if the default format is being used
+            val defaultFormat = "yyyy-MM-dd'T'HH:mm:ss[.SSS][XXX]"
+            val timestampFormat = a.options.getOrElse("timestampFormat", defaultFormat)
```

Comment:
I don't think this does 100% of what we want

https://github.com/apache/spark/blob/7120e6b88f2327ffb71c4bca14b10b15aeb26c32/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala#L125-L130

include a check for the legacyTimeParserPolicy.  Either we need to use `JSONOptions` directly to get the format that we should use, or we need to replicate the behavior exactly. I would prefer JSONOptions if possible, even through a trampoline, because it lets us know exactly what it is, even if it changes from one version of spark to another.